trio/_util.py

```diff
@@ -370,8 +370,5 @@ def name_asyncgen(agen: AsyncGeneratorType[object, t.NoReturn]) -> str:
         module = agen.ag_frame.f_globals["__name__"]
     except (AttributeError, KeyError):
         module = f"<{agen.ag_code.co_filename}>"
-    try:
-        qualname = agen.__qualname__
-    except AttributeError:
-        qualname = agen.ag_code.co_name
+    qualname = getattr(agen, "__qualname__", agen.ag_code.co_name)
```

Comment:
Not sure, but perhaps it might be better to leave this as a try-except, since then `agen.ag_code.co_name` doesn't have to be evaluated in most cases.

Reply:
Reverted with 82900e5